mr/src/main/java/org/opensearch/hadoop/rest/commonshttp/CommonsHttpTransport.java

```diff
@@ -712,12 +793,95 @@ public Object run() throws Exception {
             headerValues.add(responseHeader.getValue());
         }
 
-        // the request URI is not set (since it is retried across hosts), so use the http info instead for source
+        // the request URI is not set (since it is retried across hosts), so use the
+        // http info instead for source
         return new SimpleResponse(http.getStatusCode(), new ResponseInputStream(http), httpInfo, headers);
     }
 
+    private String queryParamsString(Multimap<String, String> queryParams) {
+        final ImmutableList.Builder<String> result = ImmutableList.builder();
+        for (Map.Entry<String, Collection<String>> param : new TreeMap<>(queryParams.asMap()).entrySet()) {
+            for (String value : param.getValue()) {
+                result.add(SdkHttpUtils.urlEncode(param.getKey()) + '=' + SdkHttpUtils.urlEncode(value));
+            }
+        }
+
+        return '?' + AMPERSAND_JOINER.join(result.build());
+    }
+
+    private void awsSigV4SignRequest(Request request, HttpMethod http, byte[] bodyBytes)
+            throws UnsupportedEncodingException {
+        String awsRegion = settings.getAwsSigV4Region();
+        String awsServiceName = settings.getAwsSigV4ServiceName();
+
+        Region signingRegion = Region.of(awsRegion);
+
+        final AwsCredentialsProvider credentials = DefaultCredentialsProvider.create();
```

Comment:
could we provide configuration also? similar to s3a,  https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html#Authenticating_with_S3

e.g.
```
opensearch.aws.credentials.provider: com.amazonaws.auth.InstanceProfileCredentialsProvider
```

Reply:
My assumption was that using the [DefaultCredentialsProvider](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/auth/credentials/DefaultCredentialsProvider.html) covers all the possible scenarios for credentials including `InstanceProfileCredentialsProvider`

Reply:
I think we should leave the decision to user.